tools/clang/unittests/SPIRV/FileTestFixture.cpp

```diff
@@ -0,0 +1,129 @@
+//===- FileTestFixture.cpp ------------- File Test Fixture Implementation -===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include <fstream>
+
+#include "FileTestFixture.h"
+#include "FileTestUtils.h"
+
+#include "effcee/effcee.h"
+
+namespace clang {
+namespace spirv {
+
+namespace {
+const char hlslStartLabel[] = "// Run:";
+const std::vector<std::string> checkCommands = {"CHECK:",       "CHECK-NEXT:",
+                                                "CHECK-SAME:",  "CHECK-DAG:",
+                                                "CHECK-LABEL:", "CHECK-NOT:"};
+
+/// \brief Checks if the given string is any of the valid effcee CHECK commands.
+bool isCheckCommand(std::string &s) {
+  return std::any_of(checkCommands.begin(), checkCommands.end(),
+                     [&s](const std::string command) {
+                       return s.compare(0, command.length(), command) == 0;
+                     });
+}
+
+} // end anonymous namespace
+
+bool FileTest::parseInputFile() {
+  bool foundRunCommand = false;
+  bool foundCheckCommands = false;
+  std::ostringstream checkCommandStream;
+  std::ifstream inputFile;
+  inputFile.exceptions(std::ifstream::failbit);
+  try {
+    inputFile.open(inputFilePath);
+    for (std::string line; !inputFile.eof() && std::getline(inputFile, line);) {
+      if (line.find(hlslStartLabel) != std::string::npos) {
+        foundRunCommand = true;
+        if (!fileTestUtils::processRunCommandArgs(line, &targetProfile,
+                                                  &entryPoint)) {
+          // An error has occured when parsing the Run command.
+          return false;
+        }
+      }
+      // Strip the leading "//" if any
+      if (line.size() > 2u && line[0] == '/' && line[1] == '/') {
+        line = line.substr(2);
+      }
+      // Skip any leading whitespace
+      size_t found = line.find_first_not_of(" \t");
+      if (found != std::string::npos) {
+        line = line.substr(found);
+      }
+
+      if (isCheckCommand(line)) {
```

Comment:
I thought about this again.  You should avoid this special parsing for check lines.  You should feed the entire input file as the checks script.  The reason is that in the failure case you want to get an accurate error report for line and column info.  The error message will only be accurate if it correlates to the original file text.